### ISSUE TITLE
Quiz : Fill blanks enhancement (score rule calculation)

### ISSUE THREAD

**Issue by tanchon** [chamilo/chamilo-lms#3236]
Hi,

The "Fill blanks" questions are my favorites ones. It's a simple but very powerful tool for the teacher. But i think that its potential is not fully exploited with the current score calculation rules.

For example with current rules :
Form : [Adam Smith||David Ricardo] is considered to be the founder of economics
Score : 10
Résult : if the student writes "Adam Smith" or "David Ricardo"=> 10 points ; if another answer => 0 point.

**Describe the solution you'd like**
It would be great if the teacher could make a finer tune of score calculation rule.

Some suggestions with fill blanks (without scroll menu) :

Form : [Adam Smith||David Ricardo] is considered to be the founder of economics
Score rule : 10||5
Result : the student writes "Adam Smith" => 10 pts ; if  "David Ricardo"=> 10 pts ; if another answer => 0 pt.

Form : [Adam Smith|||David Ricardo||Thomas Malthus||Jean Baptiste Say] is considered to be the founder of economics
Score rule : 10|||5
Result : the student writes "Adam Smith" => 10 pts ; if "David Ricardo"or "Thomas Malthus" or "Jean Baptiste Say"=> 5 pts ;  if another answer => 0 pt.

Form : [Adam Smith||David Ricardo||Thomas Malthus||Karl Marx] is considered to be the founder of economics
Score rule : 10|5|4|-3
Result : if the student writes "Adam Smith" => 10 pts ; if he writes "David Ricardo"=>5 pts .... if he writes "Karl Marx"=>-3 pts ; etc. ... ;  if another answer => 0 pt.

Form : [Adam Smith||David Ricardo||Thomas Malthus||Karl Marx||%?%] is considered to be the founder of economics
Score rule : 10|5|4|-3|-5
Result : if the student writes "Adam Smith" => 10 pts ; if  "David Ricardo"=>5 pts .... if  "Karl Marx"=>-3 pts ; etc. ... ;  if another name => -5 pts

Last, it could be generalized to **scroll menu**. For example :

Form : [Adam Smith|David Ricardo|Thomas Malthus|Karl Marx] is considered to be the founder of economics
Score rule : 10|5|4|-3
Result : if the student selects "Adam Smith" => 10 pts ; if he selects "David Ricardo"=>5 pts .... if he selects"Karl Marx"=>-3 pts.

**Additional context**
For the teacher who would like to stay with standard use, the syntax wouldn't change his habits.
For those who would like to fine-tune the score rule calculation, it would be very great.

Simple for the teacher, but certainly not for developers i presume...

Thanks !

Tanchon

**Comment by ywarnier** [comment]
Indeed, not really possible to implement in the current state of the fill in the blanks implementation.
